src/sql/workbench/services/notebook/browser/models/clientSession.ts

```diff
@@ -289,7 +291,7 @@ export class ClientSession implements IClientSession {
 			// TODO is there any case where skipping causes errors? So far it seems like it gets called twice
 			return;
 		}
-		if (connection.id !== '-1' && connection.id !== this._connectionId) {
+		if (connection.id !== '-1' && this._session && connection.id !== this._connectionId) {
```

Comment:
Could you move this session check to before/after the ID check? It's just a layout nitpick, since both of the original checks are for the ID, and checking the session in the middle looks weird.